sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+import ast
+import os
+import stat
+import re
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+# This configuration is based on vdsm.storage.lvm.LVM_CONF_TEMPLATE.
+#
+# locking_type is set to 0 in order to match lvm sos commands. With this
+# configuration we don't take any locks, so we will never block because
+# there is a stuck lvm command.
+# locking_type=0
+#
+# use_lvmetad is set to 0 in order not to show cached, old lvm metadata.
+# use_lvmetad=0
+#
+# preferred_names and filter config values are set to capture Vdsm devices.
+# preferred_names=[ '^/dev/mapper/' ]
+# filter=[ 'a|^/dev/mapper/.*|', 'r|.*|' ]
+LVM_CONFIG = """
+global {
+    locking_type=0
+    use_lvmetad=0
+}
+devices {
+    preferred_names=[ "^/dev/mapper/" ]
+    ignore_suspended_devices=1
+    write_cache_state=0
+    disable_after_error_count=3
+    filter=[ "a|^/dev/mapper/.*|", "r|.*|" ]
+}
+"""
+LVM_CONFIG = re.sub(r"\s+", " ", LVM_CONFIG).strip()
+
+
+class Vdsm(Plugin, RedHatPlugin):
+    """vdsm Plugin"""
+
+    packages = (
+        'vdsm',
+        'vdsm-client',
+    )
+
+    plugin_name = 'vdsm'
+
+    optionList = [("logsize", 'max size (MiB) to collect per log file', '',
+                   False)]
+
+    def setup(self):
+        os.environ["LVM_SYSTEM_DIR"] = "/var/run/vdsm/lvm"
+        self.add_forbidden_path('/etc/pki/vdsm/keys/*')
+        self.add_forbidden_path('/etc/pki/vdsm/libvirt-spice/*-key.*')
+        self.add_forbidden_path('/etc/pki/libvirt/private/*')
+
+        self.add_cmd_output('service vdsmd status')
+
+        self.add_copy_spec('/tmp/vds_installer*')
+        self.add_copy_spec('/tmp/vds_bootstrap*')
+        self.add_copy_spec('/etc/vdsm/*')
+
+        logsize = self.get_option('logsize')
+        if logsize is not None:
+            self.add_copy_spec('/var/log/vdsm/*', logsize)
+        else:
+            self.add_copy_spec('/var/log/vdsm/*')
+
+        self._add_vdsm_run_dir()
+        self.add_copy_spec('/usr/libexec/vdsm/hooks')
+        self.add_copy_spec('/var/lib/vdsm')
+        self.add_copy_spec('/var/log/ovirt.log')
+        self.add_copy_spec('/var/log/sanlock.log')
+
+        res = self.call_ext_prog('/usr/bin/pgrep qemu-kvm')
+        if res['status'] == 0:
+            for line in res['output']:
+                pid = line.strip()
+                self.add_copy_spec("/proc/%s/cmdline" % pid)
+                self.add_copy_spec("/proc/%s/status" % pid)
+                self.add_copy_spec("/proc/%s/mountstats" % pid)
+        self.add_cmd_output("/bin/ls -ldZ /etc/vdsm")
+        self.add_cmd_output(
+            "/bin/su vdsm -s /bin/sh -c '/usr/bin/tree -l /rhev/data-center'")
+        self.add_cmd_output(
+            "/bin/su vdsm -s /bin/sh -c '/bin/ls -lR /rhev/data-center'")
+        self.add_cmd_output(
+            "/sbin/lvm vgs -v -o +tags --config \'%s\'" % LVM_CONFIG)
```

Comment:
For all the LVM commands that use `LVM_CONFIG`, I think we would want to use the `suggest_filename` parameter to not have quite so harsh of a file name in the archive. E.G. just having `sbin_lvm_vgs_v_o_tags`.